var/spack/repos/builtin/packages/fortran/package.py

```diff
@@ -14,18 +14,28 @@ class Fortran(Package):
     homepage = "https://wg5-fortran.org/"
     virtual = True
 
-    def test(self):
+    def test_fortran(self):
+        """Compile and run 'Hello world'"""
         test_source = self.test_suite.current_test_data_dir
 
-        for test in os.listdir(test_source):
-            filepath = os.path.join(test_source, test)
-            exe_name = "%s.exe" % test
-
-            fc_exe = os.environ["FC"]
-            fc_opts = ["-o", exe_name, filepath]
+        fc_exe = os.environ["FC"]
+        fc_exe = which(join_path(self.prefix.bin, fc_exe))
+        if fc_exe is None:
+            raise SkipTest(f"{os.environ['FC']} not found in {self.version}")
 
-            compiled = self.run_test(fc_exe, options=fc_opts, installed=True)
-
-            if compiled:
-                expected = ["Hello world", "YES!"]
-                self.run_test(exe_name, expected=expected)
+        for test in os.listdir(test_source):
+            with test_part(self, f"test_fortran_{test}", f"Test {test}"):
+                filepath = os.path.join(test_source, test)
+                exe_name = f"{test}.exe"
+                fc_opts = ["-o", exe_name, filepath]
+                compiled = fc_exe(*fc_opts, output=str.split, error=str.split)
+
+                if compiled:
+                    exe = which(join_path(self.prefix.bin, exe_name))
```

Comment:
The executable built above is compiled into the current directory, not installed as `self.prefix.bin` indicates.